File: pis-frontend/src/services/PermitApplicationService.js
import axios from "./axiosConfig";

const PERMIT_API_BASE_URL = "/api/permit-applications";

/**
 * Create Permit Application without PDF
 */
export const createApplication = (applicationDto) =>
  axios.post(PERMIT_API_BASE_URL, applicationDto);

/**
 * Create Permit Application with PDF upload
 */
export const createPermitApplicationWithPdf = (applicationDto, file) => {
  const formData = new FormData();

  // IMPORTANT: must match backend @RequestPart("application")
  formData.append(
    "application",
    new Blob([JSON.stringify(applicationDto)], { type: "application/json" })
  );

  // IMPORTANT: must match backend @RequestPart("file")
  formData.append("file", file);

  return axios.post(`${PERMIT_API_BASE_URL}/with-pdf`, formData, {
    headers: {
      "Content-Type": "multipart/form-data",
    },
  });
};

/**
 * Get applications by user id
 */
export const getApplicationsByUserId = (userId) =>
  axios.get(`${PERMIT_API_BASE_URL}/user/${userId}`);

/**
 * Get all applications
 */
export const getAllApplications = () =>
  axios.get(PERMIT_API_BASE_URL);

/**
 * Get application by id
 */
export const getApplicationById = (id) =>
  axios.get(`${PERMIT_API_BASE_URL}/${id}`);

/**
 * Update application
 */
export const updateApplication = (id, applicationDto) =>
  axios.put(`${PERMIT_API_BASE_URL}/${id}`, applicationDto);

/**
 * Delete application
 */
export const deleteApplication = (id) =>
  axios.delete(`${PERMIT_API_BASE_URL}/${id}`);

File: pis-backend/src/main/java/nic/mn/pis/service/PermitApplicationService.java
package nic.mn.pis.service;

import nic.mn.pis.dto.PermitApplicationDto;
import org.springframework.web.multipart.MultipartFile;

import java.io.IOException;
import java.util.List;

public interface PermitApplicationService {

    // Create Permit Application
    PermitApplicationDto createApplication(PermitApplicationDto dto);

    // Create Permit Application with PDF Upload
    PermitApplicationDto createApplicationWithPdf(PermitApplicationDto dto, MultipartFile pdfFile) throws IOException;

    // Get All Applications (Admin/DC Use)
    List<PermitApplicationDto> getAllApplications();

    // Get Application By ID
    PermitApplicationDto getApplicationById(Long id);

    // Get Applications By UserId (Applicant Use)
    List<PermitApplicationDto> getApplicationsByUserId(Long userId);

    // Delete Application
    void deleteApplication(Long id);

    // Update Application
    PermitApplicationDto updateApplication(Long id, PermitApplicationDto dto);
}
